tools/windows-node-installer/test/e2e/utils.go

```diff
@@ -0,0 +1,79 @@
+package e2e
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/stretchr/testify/assert"
+	"reflect"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/masterzen/winrm"
+	"github.com/openshift/windows-machine-config-operator/tools/windows-node-installer/pkg/types"
+)
+
+const (
+	// winRMPort is the secure WinRM port
+	winRMPort = "5986"
+	// winFirewallCmd will verify if firewall rule is present on the windows node
+	winFirewallCmd = "powershell.exe -NonInteractive -ExecutionPolicy Bypass Get-NetFirewallRule " +
+		"-DisplayName OpenKubeletPort"
+)
+
+var (
+	// winCredentials stores the windows credentials
+	winCredentials *types.Credentials
+)
+
+// isNil is a helper functions which checks if the object is a nil pointer or not.
+func isNil(v interface{}) bool {
+	return v == nil || (reflect.ValueOf(v).Kind() == reflect.Ptr &&
+		reflect.ValueOf(v).IsNil())
+}
+
+// setupWinRMClient initializes the winRM client and executes commands on the windows node
+func setupWinRMClient(host, password, user string) (*winrm.Client, error) {
+	winRMPortNo, err := strconv.Atoi(winRMPort)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert to an integer with error: %v", err)
+	}
+	endpoint := winrm.NewEndpoint(host, winRMPortNo, true, true,
+		nil, nil, nil, time.Minute*1)
+	winRMClient, err := winrm.NewClient(endpoint, user, password)
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize winRM client with error: %v", err)
+	}
+	return winRMClient, nil
+}
+
+// runWinRMCmd executes a winRM command on the windows node.
+func runWinRMCmd(winRMClient *winrm.Client, command string) (string, string, error) {
+	stdout := new(bytes.Buffer)
+	stderr := new(bytes.Buffer)
+	exitCode, err := winRMClient.Run(command, stdout, stderr)
+	if err != nil {
+		return "", "", fmt.Errorf("failed while executing the %s remotely with error: %v", winFirewallCmd, err)
+	}
+	if exitCode != 0 {
+		return stdout.String(), stderr.String(), fmt.Errorf("%s returned %d exit code", winFirewallCmd, exitCode)
+	}
+	return stdout.String(), stderr.String(), nil
+}
+
+// runWinRMFirewallCmd checks if kubelet port is open in Windows firewall
+func runWinRMFirewallCmd() (stdout, stderr string, err error) {
+	winRMClient, err := setupWinRMClient(winCredentials.GetIPAddress(), winCredentials.GetPassword(),
+		winCredentials.GetUserName())
+	if err != nil {
+		return "", "", fmt.Errorf("failed to setup the winRMClient with error: %v", err)
+	}
+	stdout, stderr, err = runWinRMCmd(winRMClient, winFirewallCmd)
+	return stdout, stderr, err
+}
+
+// testInstanceFirewallRule checks if the created instance has opened kubelet port via firewall
+func testInstanceFirewallRule(t *testing.T) {
+	_, _, err := runWinRMFirewallCmd()
```

Comment:
Like I mentioned before, to be doubly sure, please check that stderr does not return anything. This should be done at the call site.